File: Java/Problem1.java
public class Problem1 {
  public static long sumMultiples(long m, long end) {
    long n = (end - 1L) / m;
    return m * n * (n + 1L) / 2L;
  }

  public static long sumMultiplesOf3And5(long end) {
    return sumMultiples(3L, end) + sumMultiples(5L, end) - sumMultiples(15L, end);
  }

  public static void main(String[] args) {
    System.out.println(sumMultiplesOf3And5(1_000L));
  }
}

File: Scala/Problem1.scala
object Problem1 {
  def sumMultiples(m: BigInt, end: BigInt): BigInt = {
    val n = (end - 1) / m
    m * n * (n + 1) / 2
  }

  def sumMultiplesOf3And5(end: BigInt): BigInt =
    sumMultiples(3, end) + sumMultiples(5, end) - sumMultiples(15, end)

  def main(args: Array[String]): Unit =
    println(sumMultiplesOf3And5(1000))
}
